library/Network/CloudSeeder/CommandLine.hs

```diff
@@ -1,28 +1,77 @@
 module Network.CloudSeeder.CommandLine
     ( Command(..)
-    , parseCommand
+    , ParameterSpec(..)
+    , parseArguments
+    , parseOptions
     , withInfo
     ) where
 
-import Options.Applicative (ArgumentFields, Parser, ParserInfo, Mod, subparser, strArgument, command, info, progDesc, metavar, helper)
+import Data.Monoid ((<>))
+import Options.Applicative 
+
+import qualified Data.Map as M
+import qualified Data.Set as S
 import qualified Data.Text as T
 
-data Command = DeployStack T.Text T.Text
+import Network.CloudSeeder.Types
+
+data Command = DeployStack 
+ { commandStack :: T.Text
+ , commandEnv :: T.Text
+ } deriving (Eq, Show)
```

Comment:
Given that this is sort of intended to be a sum type, I think using record syntax here is a bad idea. Why? Well, record syntax with sum types produces partial functions!

For example, imagine the following type:

```haskell
data Sum
  = Foo { foo :: Bool }
  | Bar { bar :: Int }
```

The record syntax for this type effectively generates the following two functions:

```haskell
foo :: Sum -> Bool
bar :: Sum -> Int
```

But wait! Those functions *have* to be partial. If `foo` always produces a `Bool` from a `Sum`, it must necessarily fail of `Sum` is actually a `Bar`. Indeed, that is the case:

```
ghci> foo (Bar 1)
*** Exception: No match in record selector foo
```

Therefore, I would avoid record syntax for this type.